File: 2015/go/day4/day4p1.go
// package day4 contains the solutions to the Day 4 challenge
// of the 2015 Advent of Code challenge.
package day4

import (
	"crypto/md5"
	"strconv"
	"fmt"
	"encoding/hex"
)

func Day4Part1() {
	var input = "iwrupvqb"
	index := 1
	for true {
		data := []byte(input + strconv.Itoa(index))
		output := md5.Sum(data)
		outputString := hex.EncodeToString(output[:3])
		if string(outputString[:5]) == "00000" {
			fmt.Println(index)
			return
		}
		index++
	}
	
}

File: 2015/day4/day4p1.py
import hashlib 

input = "iwrupvqb"
found = False
i = 0
while not found:
	m = hashlib.md5()
	string = (input + str(i))
	encodedStr = string.encode()
	m.update(encodedStr)
	hash = m.hexdigest()
	if hash[:5] == "00000":
		print(encodedStr)
		print(hash)
		found = True
	else:
		i += 1
print(str(i))